comp/otelcol/status/impl/status.go

```diff
@@ -0,0 +1,227 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+// Package statusimpl implements the status component interface
+package statusimpl
+
+import (
+	"embed"
+	"encoding/json"
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"io"
+	"net/http"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	statusComponent "github.com/DataDog/datadog-agent/comp/core/status"
+	ddflareextension "github.com/DataDog/datadog-agent/comp/otelcol/ddflareextension/def"
+	status "github.com/DataDog/datadog-agent/comp/otelcol/status/def"
+	apiutil "github.com/DataDog/datadog-agent/pkg/api/util"
+	"github.com/DataDog/datadog-agent/pkg/util/prometheus"
+)
+
+//go:embed status_templates
+var templatesFS embed.FS
+
+// Requires defines the dependencies of the status component.
+type Requires struct {
+	Config config.Component
+}
+
+// Provides contains components provided by status constructor.
+type Provides struct {
+	Comp           status.Component
+	StatusProvider statusComponent.InformationProvider
+}
+
+type statusProvider struct {
+	Config         config.Component
+	client         *http.Client
+	receiverStatus map[string]interface{}
+	exporterStatus map[string]interface{}
+}
+
+type prometheusRuntimeConfig struct {
+	Service struct {
+		Telemetry struct {
+			Metrics struct {
+				Readers []struct {
+					Pull struct {
+						Exporter struct {
+							Prometheus struct {
+								Host string
+								Port int
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+// NewComponent creates a new status component.
+func NewComponent(reqs Requires) Provides {
+	comp := statusProvider{
+		Config: reqs.Config,
+		client: apiutil.GetClient(false),
+		receiverStatus: map[string]interface{}{
+			"spans":           0.0,
+			"metrics":         0.0,
+			"logs":            0.0,
+			"refused_spans":   0.0,
+			"refused_metrics": 0.0,
+			"refused_logs":    0.0,
+		},
+		exporterStatus: map[string]interface{}{
+			"spans":          0.0,
+			"metrics":        0.0,
+			"logs":           0.0,
+			"failed_spans":   0.0,
+			"failed_metrics": 0.0,
+			"failed_logs":    0.0,
+		},
```

Comment:
We are keeping these two map in memory for the duration of the Agent process. Seeing that we do not update them, but rather reset them every time we display the status I think it would be interesting rather than keeping as part of the `statusProvider` struct to create them on every request. That was the garbage collector can reclaim them

Reply:
I considered doing this but not all of the prometheus metrics are guaranteed to be received by the endpoint (e.g. if the otel agent container restarts, the prometheus metric for `spans` isn't sent until the otel agent receives its first traces). In these instances, I think it would make sense to display the last reported value in the status instead of "0"

Reply:
It makes sense. Thanks for the explanation

Reply:
Yes it would. I think there may be other cases though where prometheus metrics don't get received